### ISSUE TITLE
Strange bug omitting a character

### ISSUE THREAD

**Issue by ssarantidis** [boschni/json-merger#2]
Hello,

I noticed a strange bug when executing **.mergeFile()** with the following .json file as an input:

_input.json:_

{
  "uiselect": {
    "a": "aa"
  }
}

output.json:

{
  "**iselect**": {
    "a": "aa"
  }
}

The problem is the "u" character is omitted in that case. Could you help me understand why?

Kind regards

**Comment by ssarantidis** [comment]
I managed to find the root cause of the problem. It is identified in Processor.js. I would suggest the following modification:

```
Processor.prototype._processObject = function (source, target) {
        var _this = this;
        for (var i = 0; i < this._enabledOperationNames.length; i++) {
            var name_1 = this._enabledOperationNames[i];
            var operation = this._nameOperationMap[name_1];
            var keyword = this.getKeyword(name_1);
            if (source[keyword] !== undefined) {
                this.currentScope.enterProperty(keyword);
                var result_1 = operation.processInObject(keyword, source, target);
                this.currentScope.leaveProperty();
                return result_1;
            }
        }
        if (!types_1.isObject(target)) {
            target = {};
        }
        var result = __assign({}, target);
        Object.keys(source).forEach(function (key) {
               // First check if the object's key starts with a prefix
               if (key.startsWith(_this._config.operationPrefix)) {
                var possibleKeyword = key.substr(_this._config.operationPrefix.length);
                if (possibleKeyword === "comment") {
                    return;
                }
                // call .isKeyword with "key" as an input instead of "possibleKeyword" to avoid calling twice 
                // .substr(_this._config.operationPrefix.length)
                var targetKey = _this.isKeyword(key) ? possibleKeyword : key;
                result[targetKey] = _this.processSourceProperty(source[key], key, target[key]);
            }
            // If the object's key doesn't start with a prefix 
            else{
                result[key] = _this.processSourceProperty(source[key], key, target[key]);
            }
        });
        return result;
    };
```

There were actually two problems. 1) You were not checking if the object's key starts with the prefix before checking if it contains an operation's keyword and 2) You were performing twice the key.substr(_this._config.operationPrefix.length) (first in **Processor.prototype._processObject** and second in **Processor.prototype.isKeyword** ). So in my case "uiselect" was transformed in "select" which was considered as a valid operation...

Could you please build a kit with the fix?

Thank you in advance,
Stelios

**Comment by boschni** [comment]
Good catch and thanks for debugging the issue!

The issue should be resolved in version 0.0.26.

**Comment by ssarantidis** [comment]
Thank you for the fix!
